boilerplate/app/app.tsx

```diff
@@ -35,26 +47,32 @@ interface AppProps {
  */
 function App(props: AppProps) {
   const { hideSplashScreen } = props
-  const [rootStore, setRootStore] = useState<RootStore | undefined>(undefined)
   const {
     initialNavigationState,
     onNavigationStateChange,
     isRestored: isNavigationStateRestored,
   } = useNavigationPersistence(storage, NAVIGATION_PERSISTENCE_KEY)
 
+  // load in the rootStore (which initializes it)
+  const rootStore = useStores()
+
   const [areFontsLoaded] = useFonts(customFontsToLoad)
 
-  // Kick off initial async loading actions, like loading fonts and RootStore
+  // Kick off initial async loading actions, like loading fonts and rehydrating RootStore
   useEffect(() => {
-    setupRootStore().then((rootStore) => {
-      setRootStore(rootStore)
+    ;(async () => {
+      // set up the RootStore (returns the state restored from AsyncStorage)
+      const { restoredState } = await setupRootStore(rootStore)
+
+      // reactotron integration with the MST root store (DEV only)
+      setReactotronRootStore(rootStore, restoredState)
```

Comment:
I think this is a better place to integrate Reactotron and MST than in the setupRootStore function.